telemetry/receivedenevlope.go

```diff
@@ -60,3 +60,41 @@ func (r *ReceivedEnvelope) updateProcessingError(db *sql.DB) error {
 
 	return nil
 }
+
+type SentEnvelope struct {
+	ID              int    `json:"id"`
+	MessageHash     string `json:"messageHash"`
+	SentAt          int64  `json:"sentAt"`
+	CreatedAt       int64  `json:"createdAt"`
+	PubsubTopic     string `json:"pubsubTopic"`
+	Topic           string `json:"topic"`
+	SenderKeyUID    string `json:"senderKeyUID"`
+	NodeName        string `json:"nodeName"`
+	ProcessingError string `json:"processingError"`
+	PublishMethod   string `json:"publishMethod"`
+}
+
+func (r *SentEnvelope) put(db *sql.DB) error {
+	r.CreatedAt = time.Now().Unix()
+	stmt, err := db.Prepare(`INSERT INTO sentEnvelopes (messageHash, sentAt, createdAt, pubsubTopic,
+							topic, senderKeyUID, nodeName, processingError, publishMethod)
+							VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)
+							ON CONFLICT ON CONSTRAINT sentEnvelopes_unique DO NOTHING
+							RETURNING id;`)
+	if err != nil {
+		return err
+	}
+
```

Comment:
```suggestion
defer stmt.Close()
```

hm. Interestingly enough, i see other prepared statements in the existing code not being closed :thinking:

Reply:
added in latest commit